src/microbe_stage/AgentProjectile.cs

```diff
@@ -7,22 +8,39 @@
 [SceneLoadedClass("res://src/microbe_stage/AgentProjectile.tscn", UsesEarlyResolve = false)]
 public class AgentProjectile : RigidBody, ITimedLife
 {
+    private Particles particles;
+
+    [JsonProperty]
+    private float? FadeTimeRemaining { get; set; }
+
     public float TimeToLiveRemaining { get; set; }
     public float Amount { get; set; }
     public AgentProperties Properties { get; set; }
     public Node Emitter { get; set; }
 
     public void OnTimeOver()
     {
-        Destroy();
+        BeginDestroy();
```

Comment:
I was just playtesting this and I noticed that this gets called continuously from the timed life system. Meaning that BeginDestroy needs to have
```
if(FadeTimeRemaining != null)
    return;
```
at the start, other the toxins never despawn.